lib/Drivers/STM32F4xx_HAL_Driver/Src/stm32f4xx_hal.c

```diff
@@ -173,7 +173,8 @@ HAL_StatusTypeDef HAL_Init(void)
   HAL_NVIC_SetPriorityGrouping(NVIC_PRIORITYGROUP_4);
 
   /* Use systick as time base source and configure 1ms tick (default clock after Reset is HSI) */
-  HAL_InitTick(TICK_INT_PRIORITY);
+  //HAL_InitTick(TICK_INT_PRIORITY);
+  //already initialized at this point, second initialization is causing problems
```

Comment:
This does not look right. 

I believe we should either a) not initialize the tick timer before this is called (to prevent the second initialization), or b) make HAL_InitTick detect that the timer has been initialized already and on subsequent calls do nothing. This way, we don't have to change the code in the HAL library.